### PR TITLE
Implement Scale Encode and Decode Traits for Light Client

### LINKED ISSUE #19: Implement Scale Encode and Decode Traits for Light Client
We need to implement Scale encode and decode traits for:

- [ ] `LightClientState`
- [ ] `LightClientUpdate`

### PR DESCRIPTION
**Pull request by dharjeezy**
Implements Derived types for easy scale codec encoding and decoding
closes #19

### CODE REVIEW

**Review by Wizdave97 — CHANGES_REQUESTED** [review]
We also need the reverse case from our derived types to the ethreum consensus types.

File: primitives/src/derived_types.rs [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+	pub state_root: [u8; 32],
+	pub body_root: [u8; 32],
+}
+
+impl From<bellatrix::BeaconBlockHeader> for BeaconBlockHeader {
+	fn from(beacon_block_header: bellatrix::BeaconBlockHeader) -> Self {
+		BeaconBlockHeader {
+			slot: beacon_block_header.slot,
+			proposer_index: beacon_block_header.proposer_index as u64,
+			parent_root: beacon_block_header.parent_root.as_bytes().try_into().unwrap(),
```
Comment (Wizdave97) [inline primitives/src/derived_types.rs:45]: Use `expect` instead of `unwrap()`

Comment (Wizdave97) [reply primitives/src/derived_types.rs:45]: Or better still let's implement `TryFrom`  instead of `From`

**Review by Wizdave97 — CHANGES_REQUESTED** [review]

**Comment by Wizdave97** [comment]
With `TryFrom` we can just return errors when conversion fails.

**Comment by dharjeezy** [comment]
> @dharjeezy This is still not implemented
> 
> > We also need the reverse case from our derived types to the ethereum consensus types.

Yes, i am on it.

**Review by Wizdave97 — APPROVED** [review]

**Pull request merged**